core/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -319,14 +323,26 @@ public ShardRouting activePrimary(ShardId shardId) {
     /**
      * Returns one active replica shard for the given shard id or <code>null</code> if
      * no active replica is found.
+     *
+     * Since replicas could possibly be on nodes with a newer version of ES than
+     * the primary is, this will return replicas on the lowest version of ES.
      */
     public ShardRouting activeReplica(ShardId shardId) {
```

Comment:
now that this method does not just select any active replica, maybe we should rename it to something that better reflects its purpose